Agalles/copypasta fix

AmyLGalles opened the pull request:
Fixing issue with Unicode spaces not being recognized as token terminators

AmyLGalles committed:
WIP fix that works but is inelegant

AmyLGalles committed:
setting up lita to catch UTF-8 spaces to fix copy and paste issue

edgeworthch commented on lib/lita/adapters/slack/message_handler.rb:
```diff
@@ -41,7 +41,9 @@ def handle
-            data["text"].sub(/^\s*<@#{robot_id}>/, "@#{robot.mention_name}")
+            data["text"].sub(/^\s*<@#{robot_id}>/, "@#{robot.mention_name}") #original
+            # data["text"].sub(/^/[[:space:]]/*<@#{robot_id}>/, "@#{robot.mention_name}") #never works
+            # data["text"].to_s.gsub(/[[:space:]]/,' ').sub(/^\s*<@#{robot_id}>/, "@#{robot.mention_name}") #works but not ideal
```
Instead of leaving commented-out code in the file, I'd suggest moving the various options into the git commit message, and use full paragraphs to describe the "why" of the option you chose

edgeworthch commented on lib/lita/adapters/slack/message_handler.rb:
```diff
@@ -41,7 +41,9 @@ def handle
 
```
commenting here for threading:

Should we create a new branch off of Erik's branch so it has a name that's clearer regarding why we're using that particular branch?  I'm honestly not sure how confusing the branch name would be for future engineers working on this

edgeworthch replied:
Actually, scratch that—I think it's better to stick with the changes you made to the [changebot repo](https://github.com/change/changebot/pull/41/files#diff-4220a334d4c3d81c674519e059c0a3a38dcc668d8afa679aeb8ff0f579df25d3R23) instead, and not make the changes here.  Am I missing a strong reason for making the no-op changes in this PR?

AmyLGalles replied:
I find it confusing, I imagine future engineers will as well.

AmyLGalles committed:
removing temporary commented-out configs

These were used during debugging

AmyLGalles committed:
removing commented-out configs used during debugging

            # data["text"].to_s.gsub(/[[:space:]]/,' ').sub(/^\s*<@#{robot_id}>/, "@#{robot.mention_name}") #works but not ideal

AmyLGalles committed:
Merge branch 'agalles/copypasta-fix' of github.com:change/lita-slack into agalles/copypasta-fix
